src/command_modules/azure-cli-cognitiveservices/azure/cli/command_modules/cognitiveservices/custom.py

```diff
@@ -38,6 +39,29 @@ def list_kinds(client):
     return sorted(list(kinds))
 
 
+def list_skus(cmd, kind=None, location=None, resource_group_name=None, account_name=None):
+    if resource_group_name is not None or account_name is not None:
+        logger.warning(
+            'list-skus with an existing account has been deprecated and will be removed in a future release.')
+        if resource_group_name is None:
+            # account_name must not be None
+            raise CLIError('--resource-group is required when --name is specified.')
+        # keep the original behavior to avoid breaking changes
+        return cf_accounts(cmd.cli_ctx).list_skus(resource_group_name, account_name)
+
+    # in other cases, use kind and location to filter SKUs
+    def _filter_sku(_sku):
+        if kind is not None:
+            if _sku.kind != kind:
+                return False
+        if location is not None:
+            if location.lower() not in [x.lower() for x in _sku.locations]:
+                return False
+        return True
+
+    return list(filter(_filter_sku, cf_resource_skus(cmd.cli_ctx).list()))
```

Comment:
Use list comprehension instead of `filter`:

- https://hacktec.gitbooks.io/effective-python/content/en/Chapter1/Item7.html
- https://www.artima.com/weblogs/viewpost.jsp?thread=98196

Reply:
cool, didn't know it makes differences.